Frontend_Web_Development_React_Angular_Vue/React/Week 9/react-icons.md

```diff
@@ -0,0 +1,271 @@
+        .top-react-icons{
+            color: yellow;
+        }
+        ```
+
+        `Output`
+        
+        ![Output of above](./assets/coloured-two.PNG)
+
+    - ### Styling single icon
+
+        Let's take the other two icons for this
+
+        ```js
+        <IconContext.Provider value={{ color: "blue" }}>
+        <IoLogoLinkedin />
+        </IconContext.Provider>
+        <IconContext.Provider value={{ color: "green" }}>
+        <DiGithubBadge />
+        </IconContext.Provider>
+        ```
+
+        `Output`
+
+        ![output of above](./assets/coloured-single.PNG)
+
+
+### 2. Using styled component
+
+In this tutorial, I'm assuming you know [styled component](https://styled-components.com/), if no then don't worry, our method 1 for styling icons works just fine.
+
+In case if you want to explore styled component, visit [here](https://styled-components.com/)
+
+- Install styled components
+```js
+npm install --save styled-components
+```
+
+- Import styled-component
+```js
+import styled from 'styled-components'
+```
+
+- Import some new icons or you can try on previous one
+
+```js
+import { BsFillAlarmFill } from "react-icons/bs"
+```
+
+- Use icon inside a new div
+```js
+<div>
+    <BsFillAlarmFill />
+</div>
+```
+
+- Now, go to the last line of the file and write the code to style the new icon.
+
+```js
+const Alarm = styled(BsFillAlarmFill)`
+  color: red;
+  transform: scale(2);
+  margin: 5%;
+`;
+```
+
+- Rename your `<BsFillAlarmFill>` with the name you have given in styled component i.e, `<Alarm />`
+
+```js
+<div>
+    <Alarm />
+</div>
+```
+
+- Output
+
+![Final Ourput](./assets/final-output.PNG)
+
+### Your final code inside App.js should look like this
+
+```js
+import React from "react";
+import styled from 'styled-components'
+import { IconContext } from "react-icons";
+import { FaCodepen } from "react-icons/fa";
+import { AiFillTwitterCircle } from "react-icons/ai";
+import { DiGithubBadge } from "react-icons/di";
+import { IoLogoLinkedin } from "react-icons/io";
+import { BsFillAlarmFill } from "react-icons/bs"
+import "./App.css";
+
+function App() {
+  return (
+    <div>
+      <IconContext.Provider value={{ className: "top-react-icons" }}>
+        <FaCodepen />
+        <AiFillTwitterCircle />
+      </IconContext.Provider>
+
+      <IconContext.Provider value={{ color: "blue" }}>
+        <IoLogoLinkedin />
+      </IconContext.Provider>
+      <IconContext.Provider value={{ color: "green" }}>
+        <DiGithubBadge />
+      </IconContext.Provider>
+
+      <div>
+        <Alarm />
+      </div>
+    </div>
+  );
+}
+
+export default App;
+
+const Alarm = styled(BsFillAlarmFill)`
+  color: red;
+  transform: scale(2);
+  margin: 5%;
+`;
+```
+
+## Conclusion
+Phew! We have covered so many things. Now, it's time to apply all these things in your beautiful projects. Hopw you have learnt everything about React Icons.
+
+__Contributor__ : [Harsh Anand](https://www.github.com/its-me-Harsh-Anand)
```

Comment:
Remove your name from the Documentation.

Reply:
Okay sir, I'll remove but it was compulsary to add name in JavaScript domain. So, I thought it is compulsary here too.